auth/shared-signal-receiver/handlers/request-handler.ts

```diff
@@ -6,21 +8,84 @@ import { handleAccountPurgedRequest } from './account-purged-handler';
 import type { APIGatewayProxyResult } from 'aws-lambda';
 import { generateResponse } from '../response';
 import { ReasonPhrases, StatusCodes } from 'http-status-codes';
+import { CognitoError } from '../errors';
+import { logMessages } from '../log-messages';
+import { verifyUsername } from '../cognito/verify-users';
 
 interface Handler {
   schema: z.ZodType;
   /* eslint-disable @typescript-eslint/no-explicit-any */
   handle: (data: any) => Promise<APIGatewayProxyResult>;
+  schemaName: string;
+  allowedChangeType?: string;
 }
 
 const handlers: Handler[] = [
   {
     schema: credentialChangeSchema,
     handle: handleCredentialChangeRequest,
+    schemaName:
+      'https://schemas.openid.net/secevent/caep/event-type/credential-change',
+    allowedChangeType: 'update',
+  },
+  {
+    schema: accountPurgedSchema,
+    handle: handleAccountPurgedRequest,
+    schemaName:
+      'https://schemas.openid.net/secevent/risc/event-type/account-purged',
   },
-  { schema: accountPurgedSchema, handle: handleAccountPurgedRequest },
 ];
 
+const isUserValid = async (
+  incomingRequest: any,
+  schemaName: string,
+): Promise<boolean> => {
+  const jti = incomingRequest.jti as string;
+
+  // eslint-disable-next-line security/detect-object-injection, @typescript-eslint/no-unsafe-assignment
+  const schema = (incomingRequest as { events: Record<string, any> }).events[
+    schemaName
+  ];
+
+  const username = schema.subject.uri as string;
+
+  if (!(await verifyUsername(username))) {
+    console.warn(logMessages.SIGNAL_WARN_USER_NOT_FOUND, {
+      userId: username,
+      correlationId: jti,
+    });
+    return false;
+  }
+  return true;
+};
+
+const isChangeTypeValid = (
+  incomingRequest: any,
+  schemaName: string,
+  changeType?: string,
+): boolean => {
+  if (changeType === undefined) {
+    return true;
+  }
+  const jti = incomingRequest.jti as string;
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, security/detect-object-injection
+  const schema = (incomingRequest as { events: Record<string, any> }).events[
+    schemaName
+  ];
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const incomingChangeType = schema.change_type;
+  if (incomingChangeType !== changeType) {
+    console.error(logMessages.SIGNAL_ERROR_UNKNOWN_CHANGE_TYPE, {
+      userId: schema.subject.uri as string,
+      correlationId: jti,
+      changeType: incomingChangeType as string,
+    });
+    return false;
+  }
+
+  return true;
+};
+
```

Comment:
I think this should just be part of the schema validation - i.e. Change type has to be an enum/literal of:

- create
- revoke
- update
- delete